Sources/Sentry/SentryFileManager.m

```diff
@@ -217,44 +225,72 @@ - (void)handleEnvelopesLimit
 }
 
 - (void)storeCurrentSession:(SentrySession *)session
+{
+    [self storeSession:session sessionFilePath:self.currentSessionFilePath];
+}
+
+- (void)storeCrashedSession:(SentrySession *)session
+{
+    [self storeSession:session sessionFilePath:self.crashedSessionFilePath];
+}
+
+- (void)storeSession:(SentrySession *)session sessionFilePath:(NSString *)sessionFilePath
 {
     NSData *sessionData = [SentrySerialization dataWithSession:session error:nil];
-    [SentryLog logWithMessage:[NSString stringWithFormat:@"Writing to current session: %@",
-                                        self.currentSessionFilePath]
+    [SentryLog logWithMessage:[NSString stringWithFormat:@"Writing session: %@", sessionFilePath]
                      andLevel:kSentryLogLevelDebug];
-    @synchronized(self.currentSessionFilePath) {
-        [sessionData writeToFile:self.currentSessionFilePath options:NSDataWritingAtomic error:nil];
+    @synchronized(_sessionLock) {
```

Comment:
L: We could use `sessionFilePath` as the lock object here instead. Writing two different files don't need to synchronize. Assuming `sessionFilePath` is always the same insrtances (these are fields in the class (always same reference), correct?, And they can't be null, which would anyway break here when writing.

This is true to all usages of `_sessionLock` in this file.

Reply:
Good, point. Thanks